File: Napakalaki-Ruby/lib/monster.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module NapakalakiGame

require_relative 'prize.rb'
require_relative 'bad_consequence.rb'

class Monster
  def initialize(name,level,bc,prz,ic)
      @levelChangeAgainstCultistPlayer=ic
      @name=name
      @combatLevel=level
      @badcons=bc
      @prize=prz 
  end
  
  
  def getName
    @name
  end
  
  def getCombatLevel
    @combatLevel
  end
  
  def getCombatLevelAgainstCultistPlayer
    @combatLevel+@levelChangeAgainstCultistPlayer
  end
  
  def getBadConsequence
    @badcons
  end
  
  def getLevelsGained()
    @prize.getLevels
  end
       
  def getTreasuresGained() 
    @prize.getTreasures
  end
  
  def to_s
      "#{@name}, nivel = #{@combatLevel}, mal rollo = \n #{@badcons.to_s}\n buen rollo = #{@prize.to_s}"
  end
  
  end
end

File: Napakalaki-Java/src/NapakalakiGame/Monster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;


public class Monster {
    private String name;
    private int combatLevel;
    private Prize price;
    private BadConsequence bad_c;
    int levelChangeAgainstCultistPlayer;
    
    public Monster(String name_i,int level, BadConsequence bc, Prize price_i){
        name=name_i;
        combatLevel=level;
        price=price_i;
        bad_c=bc;
        levelChangeAgainstCultistPlayer=0;
    }
    
    public Monster(String name_i, int level, BadConsequence bc, Prize price_i, int lvChangeAgainstCultistPlayer){
        name = name_i;
        combatLevel=level;
        price = price_i;
        bad_c=bc;
        levelChangeAgainstCultistPlayer=lvChangeAgainstCultistPlayer;
    }            
    
    public String getName(){
        return name;
    }
    
    public int getCombatLevel(){
        return combatLevel;
    }
    
    public int getCombatLevelAgainstCultistPlayer(){
        return getCombatLevel()+levelChangeAgainstCultistPlayer;
    }
    
    public BadConsequence getBadConsequence(){
        return bad_c;
    }
    
    public int getLevelsGained(){
        return price.getLevels();
    }
    
    public int getTreasuresGained(){
            return price.getTreasures();
    }
    
  
    
    public String toString(){
        return "Name =" + name +", Level = " + Integer.toString(combatLevel)
                + ", Prize = " + price.toString() +", Bad Consequence = " + bad_c.toString();
    }
    
    
}
